mods/beds/functions.lua

```diff
@@ -58,11 +58,17 @@ local function lay_down(player, pos, bed_pos, state, skip)
 		return
 	end
 
+	local meta
+
 	-- stand up
 	if state ~= nil and not state then
 		local p = beds.pos[name] or nil
 		beds.player[name] = nil
 		beds.bed_position[name] = nil
+
+		meta = minetest.get_meta(beds.pos[name])
```

Comment:
Judging by the following code, `beds.pos[name]` (or `p`) could be nil

Reply:
I see; do you think I should just revert to the previous code? It was working fine.